mvrx-mocking/src/main/kotlin/com/airbnb/mvrx/mocking/MavericksMocks.kt

```diff
@@ -0,0 +1,156 @@
+     * Calls to [MavericksMockPrinter.startReceiver] will be no-ops unless this is enabled.
+     *
+     * This will automatically be set when [install] is called.
+     */
+    var enableMockPrinterBroadcastReceiver: Boolean = false
+
+    /**
+     * Calls to [MockableMavericksView.provideMocks] will return empty unless this is enabled.
+     *
+     * This will automatically be set when [install] is called.
+     */
+    var enableMavericksViewMocking: Boolean = false
+
+    val mockConfigFactory: MockMavericksViewModelConfigFactory
+        get() {
+            return (MvRx.viewModelConfigFactory as? MockMavericksViewModelConfigFactory)
+                ?: error("Expecting MockMvRxViewModelConfigFactory for config factory. Make sure you have called MvRxMocks#install")
+        }
+
+    /**
+     * If the application was built with the debuggable flag enabled in its Android Manifest then
+     * this will add plugins to [MvRx] that enable working with mock State. This is useful for
+     * both manual and automated testing of development builds.
+     *
+     * This function is a shortcut instead of setting each property in this object individually.
+     * For custom control you can set properties directly instead.
+     *
+     * It is safe to call this in both debug and production
+     * builds and it will take care of the correct behavior for you.
+     *
+     * The context will be used to automatically register a broadcast receiver for each
+     * ViewModel created in the app with [ViewModelStatePrinter] so that the state printing
+     * system is automatically enabled.
+     *
+     * Calling this subsequent times will replace the plugins with new instances.
+     */
+    fun install(context: Context) {
+        val isDebuggable = context.isDebuggable()
+        install(
+            mocksEnabled = isDebuggable,
+            debugMode = isDebuggable,
+            context = context
+        )
+    }
+
+    /**
+     * Choose whether to enable [MvRx] mocking tools. This is useful for
+     * both manual and automated testing of development builds.
+     *
+     * This function is a shortcut instead of setting each property in this object individually.
+     * For custom control you can set properties directly instead.
+     *
+     * The context will be used to automatically register a broadcast receiver for each
+     * ViewModel created in the app with [ViewModelStatePrinter] so that the state printing
+     * system is automatically enabled.
+     *
+     * Calling this subsequent times will replace the plugins with new instances.
+     *
+     * @param debugMode True if debug checks should be enabled
+     * @param mocksEnabled True if ViewModel mocking should be enabled.
+     * @param context Application context. If provided this will be used to register a
+     * [ViewModelStatePrinter] for each ViewModel to support mock state printing.
+     */
+    fun install(mocksEnabled: Boolean, debugMode: Boolean, context: Context?) {
```

Comment:
wanted to make this easy to setup, since I figured most people would use the mocking (ideally). i like the extension idea

Reply:
i think it is also weird to chain them like that though, since the original MvRx setup is completely overridden and thus pointless.

I can rename this to `MockableMavericks.install...` so you would use that when using the mock library instead of the vanilla

Reply:
@elihart I like that more. The other way, it sounds like something you would add on top of Mavericks, not Mavericks itself.